docs/source/contributing.rst

```diff
@@ -33,17 +34,101 @@ If you're unsure about any of these points, please open the Pull Request anyhow
 Pull Request Process
 --------------------
 
-After you open the Pull Request, there will probably be some discussion in the comments' field of the request itself.
+After you open the Pull Request, there will probably be some discussion in the comments field of the request itself.
 
 Once the Pull Request is ready to merge, it will first be merged into our internal git system for in-house automated testing.
 
-If this process passes, it will be merged onto the public github repository.
+If this process passes, it will be merged into the public GitHub repository.
 
-Legal Part
-----------
+Example Contribution Guideline
+------------------------------
+
+Checklist
+*********
+
+* Check if your example proposal has no similarities to the project
+* Use the `Arduino style guide`_
+* Add the header to all source files
+* Add the `README.md` file
+* Add inline comments if needed
+* Test the example
+
+Header
+******
+
+All the source files must include the header with the example name and license, if applicable. You can change this header as you wish, but it will be reviewed by the community and may not be accepted.
+
+Additionally, you can add some description about the example, links to the documentation, or the author's name. Just have in mind to keep it simple and short.
+
+**Header Example**
+
+.. code-block:: arduino
+
+    /* Wi-Fi FTM Initiator Arduino Example
+
+      This example code is in the Public Domain (or CC0 licensed, at your option.)
+
+      Unless required by applicable law or agreed to in writing, this
+      software is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+      CONDITIONS OF ANY KIND, either express or implied.
+    */
+
+
+README file
+***********
+
+The **README.md** file should contain the example details.
+
+Please see the recommended **README.md** file in the `example template folder <https://github.com/espressif/arduino-esp32/tree/master/libraries/ESP32/examples/Template/ExampleTemplate>`_.
+
+Inline Comments
+***************
+
+Inline comments are important if the example contains complex algorithms or specific configurations that the user needs to change.
 
```

Comment:
_I will add one more sentence:_

Brief and clear inline comments are really helpful for the example understanding and it's fast usage.